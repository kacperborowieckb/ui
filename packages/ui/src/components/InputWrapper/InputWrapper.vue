<template>
  <div class="flex flex-col gap-1" :class="[disabled && 'opacity-60']">
    <span v-if="label" class="text-foreground">
      {{ label }}
    </span>
    <div
      class="border-border has-focus-visible:ring-offset-background has-focus-visible:ring-ring text-foreground has-focus-visible:outline-none has-focus-visible:ring-2 flex min-h-[32px] flex-wrap items-center gap-1 rounded-md border p-1 text-base ring-offset-1 transition duration-200 ease-out disabled:pointer-events-none disabled:opacity-75"
      :class="{ 'border-destructive bg-destructive/5 dark:bg-destructive/10 ring-destructive!': hasError, 'bg-muted/80 pointer-events-none': disabled }"
    >
      <slot />
    </div>
    <p v-if="hasError && errorMessage" class="text-destructive dark:brightness-150">
      {{ errorMessage }}
    </p>
  </div>
</template>

<script lang="ts">
export interface InputWrapperProps {
  label?: string
  hasError?: boolean
  errorMessage?: string
  disabled?: boolean
}
</script>

<script setup lang="ts">
defineProps<InputWrapperProps>()
</script>
