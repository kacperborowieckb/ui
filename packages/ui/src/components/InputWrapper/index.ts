export {
  default as InputWrapper,
  type InputWrapperProps,
} from './InputWrapper.vue'
