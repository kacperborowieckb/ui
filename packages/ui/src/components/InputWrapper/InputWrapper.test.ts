import { shallowMount } from '@vue/test-utils'

import InputWrapper from './InputWrapper.vue'

describe('inputWrapper', () => {
  it('should render correctly with labe', () => {
    const wrapper = shallowMount(InputWrapper, { props: { label: 'mockLabel' } })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render correctly', () => {
    const wrapper = shallowMount(InputWrapper)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render correctly when disabled', () => {
    const wrapper = shallowMount(InputWrapper, { props: { disabled: true } })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render correctly when hasError with errorMessage', () => {
    const wrapper = shallowMount(InputWrapper, { props: { hasError: true, errorMessage: 'mockErrorMessage' } })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
