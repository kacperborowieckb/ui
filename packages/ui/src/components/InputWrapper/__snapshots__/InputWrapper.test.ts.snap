// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`inputWrapper > should render correctly 1`] = `
"<div class="flex flex-col gap-1">
  <!--v-if-->
  <div class="border-border has-focus-visible:ring-offset-background has-focus-visible:ring-ring text-foreground has-focus-visible:outline-none has-focus-visible:ring-2 flex min-h-[32px] flex-wrap items-center gap-1 rounded-md border p-1 text-base ring-offset-1 transition duration-200 ease-out disabled:pointer-events-none disabled:opacity-75"></div>
  <!--v-if-->
</div>"
`;

exports[`inputWrapper > should render correctly when disabled 1`] = `
"<div class="flex flex-col gap-1 opacity-60">
  <!--v-if-->
  <div class="border-border has-focus-visible:ring-offset-background has-focus-visible:ring-ring text-foreground has-focus-visible:outline-none has-focus-visible:ring-2 flex min-h-[32px] flex-wrap items-center gap-1 rounded-md border p-1 text-base ring-offset-1 transition duration-200 ease-out disabled:pointer-events-none disabled:opacity-75 bg-muted/80 pointer-events-none"></div>
  <!--v-if-->
</div>"
`;

exports[`inputWrapper > should render correctly when hasError with errorMessage 1`] = `
"<div class="flex flex-col gap-1">
  <!--v-if-->
  <div class="border-border has-focus-visible:ring-offset-background has-focus-visible:ring-ring text-foreground has-focus-visible:outline-none has-focus-visible:ring-2 flex min-h-[32px] flex-wrap items-center gap-1 rounded-md border p-1 text-base ring-offset-1 transition duration-200 ease-out disabled:pointer-events-none disabled:opacity-75 border-destructive bg-destructive/5 dark:bg-destructive/10 ring-destructive!"></div>
  <p class="text-destructive dark:brightness-150">mockErrorMessage</p>
</div>"
`;

exports[`inputWrapper > should render correctly with labe 1`] = `
"<div class="flex flex-col gap-1"><span class="text-foreground">mockLabel</span>
  <div class="border-border has-focus-visible:ring-offset-background has-focus-visible:ring-ring text-foreground has-focus-visible:outline-none has-focus-visible:ring-2 flex min-h-[32px] flex-wrap items-center gap-1 rounded-md border p-1 text-base ring-offset-1 transition duration-200 ease-out disabled:pointer-events-none disabled:opacity-75"></div>
  <!--v-if-->
</div>"
`;
